ui/client/components/graph/TypeSuggest.js

```diff
@@ -0,0 +1,135 @@
+import React from 'react'
+import PropTypes from "prop-types";
+import AceEditor from 'react-ace';
+import _ from "lodash";
+import ProcessUtils from "../../common/ProcessUtils";
+import ActionsUtils from "../../actions/ActionsUtils";
+import {connect} from 'react-redux'
+
+import 'brace/mode/jsx';
+
+import 'brace/ext/searchbox';
+import '../../brace/mode/spel'
+import '../../brace/mode/sql'
+import '../../brace/theme/nussknacker'
+import ValidationLabels from "../modals/ValidationLabels";
+import {allValid} from "../../common/Validators";
+import TypeSuggester from "./TypeSuggester";
+import {v4 as uuid4} from "uuid";
+
+class TypeSuggest extends React.Component {
+
+  static propTypes = {
+    inputProps: PropTypes.object.isRequired,
+    fieldName: PropTypes.string,
+    validators: PropTypes.array.isRequired
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      uid: uuid4(),
+      value: ProcessUtils.humanReadableType(props.inputProps.refClazzName),
+      refClazzName: props.inputProps.refClazzName
+    };
+    this.typeSuggester = new TypeSuggester(props.typesInformation);
+  }
+
+  onChange = (newValue) => {
+    newValue = newValue.replace(/(\r\n|\n|\r)/gm, ""); // remove new line char to disable multiline input
+    const newRefClazzName = this.refClazzForValue(newValue)
+    this.setState( {
+      value: newValue,
+      refClazzName: newRefClazzName
+    })
+    // TODO we shouldn't set refClazzName but only humanReadableType such String, this should be moved to BE (class TypingUtils)
+    this.props.inputProps.onValueChange({typ: {refClazzName: newRefClazzName}})
+  }
+
+  refClazzForValue = (newValue) => {
+    const typesMatchedForValue = this.typeSuggester.matchesFor(newValue);
+    return typesMatchedForValue.length === 1 ? typesMatchedForValue.head.clazzName.refClazzName : null
```

Comment:
it's not scala, there's no head :P 
typesMatchedForValue[0]

Reply:
Better will be _.get(typesMatchedForValue, '0.clazzName.refClazzName')